includes/notes/class-wc-stripe-upe-stripelink-note.php

```diff
@@ -0,0 +1,100 @@
+ */
+
+use Automattic\WooCommerce\Admin\Notes\NoteTraits;
+use Automattic\WooCommerce\Admin\Notes\Note;
+use Automattic\WooCommerce\Admin\Notes\WC_Admin_Note;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WC_Stripe_UPE_StripeLink_Note
+ */
+class WC_Stripe_UPE_StripeLink_Note {
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-stripe-upe-stripelink-note';
+
+	/**
+	 * Link to enable the UPE in store.
+	 */
+	const NOTE_DOCUMENTATION_URL = 'https://woocommerce.com/document/stripe/#section-11';
+
+	/**
+	 * Get the note.
+	 */
+	public static function get_note() {
+		$note_class = self::get_note_class();
+		$note       = new $note_class();
+
+		$note->set_title( __( 'Increase conversion at checkout', 'woocommerce-gateway-stripe' ) );
+		$note->set_content( __( 'Reduce cart abandonment and create a frictionless checkout experience with Link by Stripe. Link autofills your customer’s payment and shipping details so they can check out in just six seconds with the Link optimized experience. That’s 9x faster than shoppers who don’t use Link. Link increases conversion rates by over 7% for logged-in Link customers.', 'woocommerce-gateway-stripe' ) );
+
+		$note->set_type( $note_class::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_name( self::NOTE_NAME );
+		$note->set_source( 'woocommerce-gateway-stripe' );
+		$note->add_action(
+			self::NOTE_NAME,
+			__( 'Enable in your store', 'woocommerce-gateway-stripe' ),
+			self::NOTE_DOCUMENTATION_URL,
+			$note_class::E_WC_ADMIN_NOTE_UNACTIONED,
+			true
+		);
+
+		return $note;
+	}
+
+	/**
+	 * Get the class type to be used for the note.
+	 *
+	 * @return string
+	 */
+	private static function get_note_class() {
+		if ( class_exists( 'Automattic\WooCommerce\Admin\Notes\Note' ) ) {
+			return Note::class;
+		} else {
+			return WC_Admin_Note::class;
+		}
+	}
+
+	/**
+	 * Init Link payment method notification
+	 *
+	 * @param WC_Stripe_UPE_Payment_Gateway $gateway
+	 *
+	 * @return void
+	 * @throws \Automattic\WooCommerce\Admin\Notes\NotesUnavailableException
+	 */
+	public static function init( WC_Stripe_UPE_Payment_Gateway $gateway ) {
+		if ( ! WC_Stripe_Feature_Flags::is_upe_checkout_enabled() ) {
+			return;
+		}
+
+		// Check if Link payment is available.
+		$available_upe_payment_methods = $gateway->get_upe_available_payment_methods();
+
+		if ( ! in_array( WC_Stripe_UPE_Payment_Method_Link::STRIPE_ID, $available_upe_payment_methods, true ) ) {
+			return;
+		}
+
+		// Retrieve enabled payment methods at checkout.
+		$enabled_payment_methods = $gateway->get_upe_enabled_at_checkout_payment_method_ids();
+
+		// If card payment method is not enabled, skip. If Link payment method is enabled, skip.
+		if (
+			! in_array( WC_Stripe_UPE_Payment_Method_CC::STRIPE_ID, $enabled_payment_methods, true ) ||
+			in_array( WC_Stripe_UPE_Payment_Method_Link::STRIPE_ID, $enabled_payment_methods, true )
+		) {
+			return;
+		}
+
```

Comment:
Should we also add a condition here on the store currency, since Link only works if the store has USD? I'm not sure if it works only with the main store currency set as USD, or it allows USD from a Multicurrency plugin as well, but we could check on that "Requires currency" condition, and apply the same condition here.

<img width="533" alt="WooCommerce Stripe Link requires currency" src="https://user-images.githubusercontent.com/5509901/197618282-bfdbc7b2-8137-4a2e-973e-7f93e6c84438.png">

**Edit:** After testing, the note appears even after I have Stripe Link enabled but still requires currency.

Reply:
I added the store currency validation in [aff006b](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2462/commits/aff006b35025968d7795cde588ec263fb5f3037c) . thx